src/main/java/org/opentripplanner/graph_builder/module/osm/OpenStreetMapModule.java

```diff
@@ -897,50 +897,128 @@ private void buildElevatorEdges(Graph graph) {
+                ArrayList<Vertex> onboardVertices = new ArrayList<Vertex>();
+                for (int i = 0; i < nodes.size(); i++) {
+                    Long node = nodes.get(i);
+                    OsmVertex sourceVertex = intersectionNodes.get(node);
+                    String sourceVertexLabel = sourceVertex.getLabel();
+                    String levelName = elevatorWay.getId() + " / " + i;
+                    getOnboardVertex(
+                            graph, onboardVertices, sourceVertex, sourceVertexLabel, levelName
+                    );
                 }
 
-                // -1 because we loop over onboardVertices two at a time
-                for (Integer i = 0, vSize = onboardVertices.size() - 1; i < vSize; i++) {
-                    Vertex from = onboardVertices.get(i);
-                    Vertex to = onboardVertices.get(i + 1);
-
-                    // default permissions: pedestrian, wheelchair, and bicycle
-                    boolean wheelchairAccessible = true;
-                    StreetTraversalPermission permission = StreetTraversalPermission.PEDESTRIAN_AND_BICYCLE;
-                    // check for bicycle=no, otherwise assume it's OK to take a bike
-                    if (node.isTagFalse("bicycle")) {
-                        permission = StreetTraversalPermission.PEDESTRIAN;
-                    }
-                    // check for wheelchair=no
-                    if (node.isTagFalse("wheelchair")) {
-                        wheelchairAccessible = false;
-                    }
+                int travelTime = elevatorWay.hasTag("duration") ?
+                        Integer.parseInt(elevatorWay.getTag("duration")) :
+                        -1;
+                int levels = nodes.size();
+                boolean wheelchairAccessible = true;
+                if (elevatorWay.isTagFalse("wheelchair")) {
+                    wheelchairAccessible = false;
+                }
+                createElevatorHopEdges(
+                        onboardVertices,
+                        wheelchairAccessible,
+                        elevatorWay.isTagTrue("bicycle"),
+                        levels,
+                        travelTime
+                );
+                LOG.debug("Created elevatorHopEdges for way {}", elevatorWay.getId());
+            }
+        }
 
-                    // The narrative won't be strictly correct, as it will show the elevator as part
-                    // of the cycling leg, but I think most cyclists will figure out that they
-                    // should really dismount.
-                    ElevatorHopEdge foreEdge = new ElevatorHopEdge(from, to, permission);
-                    ElevatorHopEdge backEdge = new ElevatorHopEdge(to, from, permission);
-                    foreEdge.wheelchairAccessible = wheelchairAccessible;
-                    backEdge.wheelchairAccessible = wheelchairAccessible;
+        private void getOnboardVertex(
```

Comment:
```suggestion
        private void createElevatorVertices(
```

Previously onboard vertices were for being onboard transit vehicles, it would be good to be more explicit here. 

Also this does not return anything, so it is not a getter, but more of a generator. Finally, it creates both onboard and off-board vertices, so a better name would be suitable.

Reply:
Valid point.